include/onnxruntime/core/framework/tensor_shape.h

```diff
@@ -40,6 +40,9 @@ class TensorShape : private std::vector<int64_t> {
 
   TensorShape(const std::vector<int64_t>& dims, size_t start, size_t end);
 
+  template<typename Iter>
+  TensorShape(Iter begin, Iter end) : std::vector<int64_t>(begin, end) {}
+
```

Comment:
nit: Do we need this? std::vector has iterator based construction and we have a move ctor from an instance of std::vector<int64_t>.

Reply:
This was removed.